feat: add React text (#86)
lamalab-org/chembench-paper: src/tex/appendix.tex

However, since some subjects are composed of questions from different sources, the ranking of the models is, in some instances, different from the one on \chembenchmini.

\begin{table}

>     \caption{\textbf{Performance of the models on the \chembench corpus.} The table shows the fraction of questions answered correctly by the models for different skills and difficulty levels. Models with \enquote{T-one} in the name were run for Temperature=1, which allow to study the temperature effect in the benchmark. Systems with \enquote{ReAct} in the name are tool augmented, i.e., they can call external tools such as web search or a calculator to better answer the questions. However, we limit those systems to a maximum of ten calls to the LLM. This constraint led the systems to often not find the correct answer within the specified number of calls. In this case, we consider the answer as incorrect (see \Cref{sec:react-environment}).}

Reviewer: ```suggestion
    \caption{\textbf{Performance of the models on the \chembench corpus.} The table shows the fraction of questions answered correctly by the models for different skills and difficulty levels. Models with \enquote{T-one} in the name were run for a temperature of 1, which allows us to study the temperature effect in the benchmark. Systems with \enquote{ReAct} in the name are tool augmented, i.e., they can call external tools such as web search or a calculator to answer the questions better. However, we limit those systems to ten calls to the \gls{llm}. This constraint often led the systems to not find the correct answer within the specified number of calls. In this case, we consider the answer as incorrect (see \Cref{sec:react-environment}).}
```

    \label{fig:tool_use}
\end{figure}

\clearpage
\subsection{Tool augmented models} \label{sec:react-environment}

> In addition to directly prompting LLMs, we also investigated the performance of tool-augmented systems.

Reviewer: ```suggestion
In addition to directly prompting \glspl{llm}, we also investigated the performance of tool-augmented systems.
```

> For this, we investigated the two models which showed the best overall results, \GPTFourO and \ClaudeThreeFiveSonnet (\oone overperformed both models but is not recommended using this model with such \enquote{reasoning} prompts such as ReAct\autocite{yao2022react} or CoT\autocite{wei2023cot}). 

Reviewer: ```suggestion
For this, we investigated the two models which showed the best overall results, \GPTFourO and \ClaudeThreeFiveSonnet (\oone overperformed both models but is not recommended using this model with such \enquote{reasoning} prompts such as ReAct\autocite{yao2022react} or \gls{cot}\autocite{wei2023cot}). 
```

For both models, we created a ReAct-style tool augmentation environment in which models had access to WolframAlpha, the ArXiv API, the Wikipedia access, and a web search (using Brave browser API).

> We based the election of these tools based in the most used tools by the humans (see \Cref{fig:tool_use}).

Reviewer: ```suggestion
We based the selection of these tools on the most used tools by humans (see \Cref{fig:tool_use}).
```

> Additionally, we added two specific tools to convert IUPAC names to SMILES, and SMILES to IUPAC names.

Reviewer: ```suggestion
Additionally, we added two specific tools to convert \gls{iupac} names to \gls{smiles}, and \gls{smiles} to \gls{iupac} names.
```

> These conversion tools allow us to better understand how the agents perform for some specific questions in this agent environment configurations.

Reviewer: ```suggestion
These conversion tools allow us to better understand how the agents perform for some specific questions in this agent environment configuration.
```

> We implemented the systems using Langchain with the default ReAct prompt and constrained the system to a maximum of ten LLM calls.

Reviewer: ```suggestion
We implemented the systems using Langchain with the default ReAct prompt and constrained the system to a maximum of ten \gls{llm} calls.
```

is there a good citation for langchain?

Reviewer 2: their git repo: https://github.com/langchain-ai/langchain/blob/master/CITATION.cff

> While for \ClaudeThreeFiveSonnet the overall performance increased (less than 1\%), they show a sloped decrease for \GPTFourO compared with the \llm without tools (\Cref{tab:performance_table}). 

Reviewer: ```suggestion
While for \ClaudeThreeFiveSonnet the overall performance increased (by less than 1\%), we observe a decrease in performance for \GPTFourO compared with the \llm without tools (\Cref{tab:performance_table}). 
```

Reviewer 2: @MrtinoRG as far as I see there is no improvement for Claude? it is almost the same number, without the agent being slightly better?

> While for \ClaudeThreeFiveSonnet the overall performance increased (by less than 1\%), we observe a decrease in performance for \GPTFourO compared with the \llm without tools (\Cref{tab:performance_table}). 

Reviewer 2: ```suggestion
While for \ClaudeThreeFiveSonnet the overall performance did not change, we observe a decrease in performance for \GPTFourO compared with the \llm without tools (\Cref{tab:performance_table}). 
```

> If we study the results by each type of question, we observe an improvement for questions regarding electron count or point group questions.

Reviewer: ```suggestion
If we study the results by each type of question, we observe an improvement for questions regarding electron counts or point groups of compounds.
```

> On the other hand, the scores decreased question asking about the number of isomers or information contained in pictograms.

Reviewer: ```suggestion
However, the scores decreased for questions asking about the number of isomers or \gls{ghs} pictograms.
```

there was no "on the one hand"

> For the specific questions about converting IUPAC names to SMILES, the results decreased notably despite the specific tools that were prepared for those questions. 

Reviewer: ```suggestion
For the specific questions about converting \gls{iupac} names to \gls{smiles}, the results decreased notably despite the models having access to specific tools prepared for those questions. 
```
